### PR TITLE
add test for handlertorch

### PR DESCRIPTION
**Pull request by suzannejin**
- [x] a pytest test for handletorch is added as an adaption from the previous unittest version (now deprecated).
- [x] make format to improve formatting etc
- [x] modified ruff.toml configurations to not run linting on unittest fiiles

### CODE REVIEW

File: tests/test_csv_processing.py
Comment (alessiovignoli) [inline tests/test_csv_processing.py:1]: invert args and attributes order in docstring

**Review by alessiovignoli — COMMENTED** [review]
transform handletorch tests from just checking input and output shapes to aqctually identicity of matrices. MEaning shape and content, this is a more stringent test that requires also the shape to be exact.

File: config/ruff.toml [hunk 1]
```diff
@@ -63,6 +63,7 @@ ignore = [
     "PLR2004",  # Magic value used in comparison
     "S101",  # Use of assert detected
 ]
+"tests/unittest_*.py" = ["ALL"]
```
[the remark this answers is not in the document]
Comment (mathysgrapotte) [reply config/ruff.toml:66]: do you mean temporary until unittests are converted to pytest ?

File: tests/test_data_transformers.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ class DataTransformerTest:
         expected_multiple_outputs: The expected outputs for the multiple inputs.
     """
 
-    def __init__( # noqa: D107
+    def __init__(  # noqa: D107
```
Comment (mathysgrapotte) [inline tests/test_data_transformers.py:29]: what does # noqa: D107 mean ?

Comment (suzannejin) [reply tests/test_data_transformers.py:29]: don;t know, igor put it
but it is probably a line that is used to remove a given from certain linting checking

**Comment by suzannejin** [comment]
> transform handletorch tests from just checking input and output shapes to aqctually identicity of matrices. MEaning shape and content, this is a more stringent test that requires also the shape to be exact.

hello, so now i have modified the test so that it alternatively loads the data into the expected values. And then check if these values are the same as the values loaded directly using the corresponding class.

**Review by alessiovignoli — APPROVED** [review]
very nicely done. This could become the standard of future pytests.

**Pull request merged**